### PR TITLE
Do mutate shared state with the old SourceMapGenerator after calling …

### LINKED ISSUE #191: Error in 0.4.3: "0" is not in the set.
I update source-map in PostCSS to 0.4.3 and got a errors in tests on `map.toString()`

```
Error: "0" is not in the set.
```

/cc @fitzgen

### PR DESCRIPTION
**Pull request by fitzgen**
…SourceMapConsumer.fromSourceMap.

Fixes #191.

r? @jlongster

### CODE REVIEW

**Comment by fitzgen** [comment]
Woops, left a require('...') call in there that shouldn't have been in there. Force pushed a fix.

File: lib/source-map/source-map-consumer.js [hunk 1]
```diff
@@ -337,25 +337,45 @@ define(function (require, exports, module) {
...
 
-          if (m.name !== null) {
-            m.name = smc._names.indexOf(m.name);
+      for (var i = 0, length = generatedMappings.length; i < length; i++) {
+        var srcMapping = generatedMappings[i];
+        var destMapping = new Mapping;
+        destMapping.generatedLine = srcMapping.generatedLine;
+        destMapping.generatedColumn = srcMapping.generatedColumn;
+
+        if (srcMapping.source) {
```
Comment (jlongster) [inline lib/source-map/source-map-consumer.js]: Any reason you dropped the explicit `!== null` checks? I feel like I remember you adding them somewhere else for JIT reasons, I could be completely wrong though.

**Comment by jlongster** [comment]
I wonder if this will make pretty printing slower, but it seems like the right thing to do so r+

**Comment by fitzgen** [comment]
Just fixed up the commit message.

**Pull request merged**